internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -1073,3 +1082,67 @@ func (r *ClusterDeploymentsReconciler) ensureOwnerRef(ctx context.Context, cd *h
 	}
 	return r.Update(ctx, ci)
 }
+
+func (r *ClusterDeploymentsReconciler) clusterControllerLogs(
+	clusterInstall *hiveext.AgentClusterInstall,
+	cluster *common.Cluster) error {
+
+	var status corev1.ConditionStatus
+	var reason, message string
+	switch {
+	case !swag.IsZero(cluster.ControllerLogsCollectedAt):
+		status = corev1.ConditionTrue
+		reason = LogCollectionCompletedReason
+		downloadURL, err := r.getControllerLogsDownloadURL(cluster)
+		if err != nil {
+			return err
+		}
+		message = fmt.Sprintf("%s - download URL: %s", LogCollectionCompletedMsg, downloadURL)
```

Comment:
The URL should not be in the message but in a new field under AgentClusterInstall Status.
debugInfo:
    logsURL: http://10.1.37.42:6000/api/assisted-install/v1/.........